src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseevents/RespondentResponseDeadlineCallbackHandlerTest.java

```diff
@@ -0,0 +1,107 @@
+import uk.gov.hmcts.reform.civil.service.FeatureToggleService;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.handler.callback.camunda.caseevents.RespondentResponseDeadlineCallbackHandler.TASK_ID;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {
+    RespondentResponseDeadlineCallbackHandler.class,
+    DeadlinesCalculator.class,
+    JacksonAutoConfiguration.class
+})
+public class RespondentResponseDeadlineCallbackHandlerTest extends BaseCallbackHandlerTest {
+
+    @Autowired
+    private RespondentResponseDeadlineCallbackHandler handler;
+
+    @MockBean
+    private FeatureToggleService featureToggleService;
+
+    @Autowired
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @MockBean
+    private DeadlinesCalculator deadlinesCalculator;
+
+
+    @Test
+    void shouldUpdateRespondent1ResponseDeadlineTo28days_whenClaimIssueTimeIsBefore4pm() {
+
+        LocalDateTime localDateTime = LocalDateTime.of(2023, 10, 30, 12, 0, 0);
+        when(deadlinesCalculator.plus28DaysAt4pmDeadline(any())).thenReturn(localDateTime);
+        when(featureToggleService.isLipVLipEnabled()).thenReturn(true);
+        CaseData caseData = CaseDataBuilder.builder()
+            .issueDate(LocalDate.of(2023, 10, 2))
+            .build();
+        CallbackParams params = callbackParamsOf(caseData, ABOUT_TO_SUBMIT);
+
+        // When
+        var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+        CaseData updatedData = mapper.convertValue(response.getData(), CaseData.class);
+
+        // Then
+        assertThat(updatedData.getRespondent1ResponseDeadline()).isEqualTo(localDateTime);
+
+    }
+
+    @Test
+    void shouldUpdateRespondent1ResponseDeadlineTo28days_whenClaimIssueTimeIsAfter4pm() {
+
+        LocalDateTime localDateTime = LocalDateTime.of(2023, 10, 31, 17, 0, 0);
+        when(deadlinesCalculator.plus28DaysAt4pmDeadline(any())).thenReturn(localDateTime);
+        when(featureToggleService.isLipVLipEnabled()).thenReturn(true);
+        CaseData caseData = CaseDataBuilder.builder()
+            .issueDate(LocalDate.of(2023, 10, 2))
+            .build();
+        CallbackParams params = callbackParamsOf(caseData, ABOUT_TO_SUBMIT);
+
+        // When
+        var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+        CaseData updatedData = mapper.convertValue(response.getData(), CaseData.class);
+
+        // Then
+        assertThat(updatedData.getRespondent1ResponseDeadline()).isEqualTo(localDateTime);
+    }
+
+    @Test
+    void shouldUpdateRespondent1ResponseDeadlineToNull_whenClaimIssueDateIsNull() {
+        CaseData caseData = CaseDataBuilder.builder().build();
```

Comment:
perhaps this test case should say should not updae respondent1 response deadline when lip v lip is not enabled? as this  is what it is testing